src/agent/h1-proxy-agent.ts

```diff
@@ -79,9 +79,11 @@ export class HttpProxyAgent extends http.Agent {
     constructor(options: AgentOptions) {
         super(options);
 
+        // @ts-ignore New @types/node patch has narrower types than got-scraping exported interfaces
```

Comment:
are we sure this helps? those ignore comments wont be preserved in the compiled code, its weird we'd need those

if its about bumping `@types/node`, why dont we just bump them in this project too? and why its a problem for newly installed projects that already use the latest version of the node types?

Reply:
With `createConnection` and other methods now being parts of the `@types/node` interface, our [original overrides](https://github.com/apify/got-scraping/pull/164/files#diff-901b51e64729e4646b1c2d5203e893839ab696bbe9af423e18fb4078740f71b7R160) are throwing TS errors. We cannot change the interface of our overriding methods, since that would be breaking (we're exporting those custom agent implementations - see #151 ). 

Imo we can either stay on the old `@types/node` version and just add the methods to `wrapped-agent` (likely with some `expect-error`s there), or update the project to the latest types, but that will likely require `expect-error` here.

Reply:
i'd bump it so we are not adding ignore comments in the blind. but i am still a bit concerned about the fact that they wont make it to the compiled code - its fine for those inside implementations, but it could be a problem for those on the interface level (so over methods)

Reply:
we could preserve the ignore comments like we do in crawlee (via postbuild script), given we dont really want to maintain the project, i'd be fine with that

also i'd mention impit in the readme to atract more people to it as the successor of got scraping

Reply:
My bad, the agents are only exported through the `getAgents` method, which returns `Promise<Agents>`:

![image](https://github.com/user-attachments/assets/6c579e8e-8afa-4711-9f25-a78a260c71c6)

So this `expect-error` already takes care of the type mismatch, and users only see the "correct" interface (even though it might not be implemented correctly, but that has always been the case).

![image](https://github.com/user-attachments/assets/61ccfc62-5ea6-450a-bdca-e1c41a2e5445)

+1 for mentioning `impit`, it's high time to ditch `got-scraping`